### PR TITLE
Better exit navigation for my-ingress

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Currently, all exit navigations use history.back()

Navigate to the add-on page if:
- Add-on does not support ingress
- Add-on is not running
- Add-on is not installed

Navigate to the add-on store if the add-on is not found.

Navigate to the add-on 

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/ingress-view/hassio-ingress-view.ts [hunk 1]
```diff
@@ -97,16 +97,30 @@ class HassioIngressView extends LitElement {
...
+          navigate(`/hassio/addon/${addonInfo.slug}/info`, { replace: true });
+        } else if (!addonInfo.ingress) {
           await showAlertDialog(this, {
             text: this.supervisor.localize("my.error_addon_no_ingress"),
             title: addonInfo.name,
           });
           await nextRender();
-          history.back();
+          navigate(`/hassio/addon/${addonInfo.slug}/info`, { replace: true });
+        } else if (addonInfo.state !== "started") {
```
Comment (bramkragten) [inline hassio/src/ingress-view/hassio-ingress-view.ts:117]: This is already handled on the ingress page right?

Comment (ludeeus) [reply hassio/src/ingress-view/hassio-ingress-view.ts:117]: I guess

File: hassio/src/ingress-view/hassio-ingress-view.ts [hunk 2]
```diff
@@ -97,16 +97,23 @@ class HassioIngressView extends LitElement {
...
+          await nextRender();
+          navigate(`/hassio/addon/${addonInfo.slug}/info`, { replace: true });
+        } else if (!addonInfo.ingress) {
           await showAlertDialog(this, {
             text: this.supervisor.localize("my.error_addon_no_ingress"),
             title: addonInfo.name,
           });
           await nextRender();
-          history.back();
+          navigate(`/hassio/addon/${addonInfo.slug}/info`, { replace: true });
```
Comment (balloob) [inline hassio/src/ingress-view/hassio-ingress-view.ts:116]: If it has a web ui, should we go there?

Comment (balloob) [reply hassio/src/ingress-view/hassio-ingress-view.ts:116]: I wonder now if we should build such logic into the ingress vote actually so it works when people have a panel too

Comment (ludeeus) [reply hassio/src/ingress-view/hassio-ingress-view.ts:116]: Even if it has a webUI, that experience is not great, it only works locally

Comment (bramkragten) [reply hassio/src/ingress-view/hassio-ingress-view.ts:116]: In my setup it never works, as my HA is behind a reverse proxy 🤷‍♂️

**Review by balloob — APPROVED** [review]

**Pull request merged**